backends/dialogs/macosx/macosx-dialogs.mm

```diff
@@ -128,22 +128,22 @@ - (IBAction) showHiddenFiles : (id) sender {
 
 @end
 
-Common::DialogManager::DialogResult MacOSXDialogManager::showFileBrowser(const char *title, Common::FSNode &choice, bool isDirBrowser) {
+Common::DialogManager::DialogResult MacOSXDialogManager::showFileBrowser(const Common::U32String &title, Common::FSNode &choice, bool isDirBrowser) {
 
 	DialogResult result = kDialogCancel;
 
 	// Get current encoding
 #ifdef USE_TRANSLATION
-	CFStringRef encStr = CFStringCreateWithCString(NULL, TransMan.getCurrentCharset().c_str(), kCFStringEncodingASCII);
+	CFStringRef encStr = CFStringCreateWithCString(NULL, TransMan.getCurrentCharset().c_str() == "UTF-32" ? "UTF-8" : "ASCII", kCFStringEncodingASCII);
 	CFStringEncoding stringEncoding = CFStringConvertIANACharSetNameToEncoding(encStr);
 	CFRelease(encStr);
 #else
 	CFStringEncoding stringEncoding = kCFStringEncodingASCII;
 #endif
```

Comment:
This whole block is overly complicated. You are using `U32String` and calling `encode()`, so we know the text is in UTF-8. And ASCII is also valid UTF-8 so there is no need to make a distinction.

Thus you could just be using
`CFStringEncoding stringEncoding = kCFStringEncodingUTF8;`